### PR TITLE
[web3go] Add new package

### PR DESCRIPTION
**Pull request by lucifer1004**

### CODE REVIEW

**Comment by lucifer1004** [comment]
@giordano This package also does not build on Azure, with similar error message to #5190:

```
Evaluating: and(ne(variables['projplatforms'], ''), ne(variables['projplatforms'], '{ }'))
Expanded: and(ne('', ''), ne(variables['projplatforms'], '{ }'))
Result: False
```

**Comment by lucifer1004** [comment]
@giordano Built issue fixed.

File: W/web3go/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+sources = [
+    ArchiveSource("https://github.com/gochain/web3/archive/refs/tags/v$(version).tar.gz", "4797b4bce370526da0fb04ca2350c36ac5d9894b1cb92a24524d2826ab0c34d6")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/web3-*/
+install_license LICENSE
+CGO_ENABLED=1 make
+mkdir -p $WORKSPACE/destdir/bin && mv web3* $WORKSPACE/destdir/bin
```
Comment (giordano) [inline W/web3go/build_tarballs.jl:18]:
```suggestion
install -Dvm "web3${exeext}" "${bindir}/web3${exeext}"
```

File: W/web3go/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "freebsd"; ),
+    Platform("i686", "windows"; )
+]
```
Comment (giordano) [inline W/web3go/build_tarballs.jl:36]: This is a bizarre set of platforms.  At a quick glance missing are macOS and 64-bit Windows?  I don't see any correlation between them.

Comment (lucifer1004) [reply W/web3go/build_tarballs.jl:36]: macOS and 64-bit Windows were excluded due to different compilation issues.

**Pull request merged**